OWA: Outlook.com: Unexpected end of JSON input

### Reproduction
* Set up an @outlook.com personal account
* Choose protocol OWA
* Click on Inbox
* Login

### Actual result
* Login appears to be working (not sure)
* Error: "Unexpected end of JSON input"
* Stack: 
![image](https://github.com/user-attachments/assets/023b8f55-af7f-4c8c-b94b-805885ec3ec4)

### Expected result
* No error
* Login works

### Scope
* Business accounts (neil@beonex.on...) seem to work, without the error

I also just saw error "Unexpected token '<', "<!DOCTYPE "... is not valid JSON". Stack shows `markRead()` on an OWA account. This happened after I had the app running idle for a a day.

```stacktrace
error.ts:6 Error: Unexpected token '<', "<!DOCTYPE "... is not valid JSON
    at message.js:153:16
    at new Promise (<anonymous>)
    at WSCall.makeCall (message.js:148:12)
    at JPCWebSocket.callRemote (protocol.js:161:31)
    at Object.fetchJSON (obj.js:198:55)
    at OWAAccount.callOWA (OWAAccount.ts:179:50)
    at OWAFolder.folderCountsChanged (OWAFolder.ts:78:37)
    at OWAFolder.listMessages (OWAFolder.ts:92:21)
    at async loadFolder (MailApp.svelte:40:28)
```

--- a/app/logic/Mail/OWA/OWAAccount.ts
+++ b/app/logic/Mail/OWA/OWAAccount.ts
@@ -232,7 +232,8 @@ export class OWAAccount extends MailAccount {
     try {
       let url = this.url + "ev.owa2?ns=PendingRequest&ev=FinishNotificationRequest&UA=0";
       let response = await appGlobal.remoteApp.OWA.fetchJSON(this.partition, url);
-      let cid = response.json.cid;
+      let json = response.json;
+      let cid = json ? json.cid : null;
       // This loop only ends by exception (e.g. logout) or app shutdown.
       while (true) {
         url = this.url + "ev.owa2?ns=PendingRequest&ev=PendingNotificationRequest&UA=0&cid=" + cid + "&X-OWA-CANARY=";

--- a/backend/owa.ts
+++ b/backend/owa.ts
@@ -139,9 +139,8 @@ export async function fetchJSON(partition: string, url: string, action: string, 
   try {
     result.json = await response.json();
   } catch (ex) {
-    if (ex instanceof SyntaxError &&
-      ex.message == "Unexpected end of JSON input") {
-    } else {
+    if (ex !instanceof SyntaxError &&
+      ex.message != "Unexpected end of JSON input") {
       throw ex;
     }
   }

--- a/backend/owa.ts
+++ b/backend/owa.ts
@@ -136,7 +136,15 @@ export async function fetchJSON(partition: string, url: string, action: string, 
   result.statusText = response.statusText;
   result.url = response.url;
   result.contentType = response.headers.get('Content-Type');
-  result.json = await response.json();
+  try {
+    result.json = await response.json();
+  } catch (ex) {
+    if (ex instanceof SyntaxError &&
+      ex.message == "Unexpected end of JSON input") {
+    } else {
+      throw ex;
+    }
+  }
   return result;
 }
 

--- a/backend/owa.ts
+++ b/backend/owa.ts
@@ -139,8 +139,11 @@ export async function fetchJSON(partition: string, url: string, action: string, 
   try {
     result.json = await response.json();
   } catch (ex) {
-    if (ex !instanceof SyntaxError &&
-      ex.message != "Unexpected end of JSON input") {
+    if (ex instanceof SyntaxError &&
+        (ex.message == "Unexpected end of JSON input" ||
+         ex.message?.includes("DOCTYPE"))) {
+      response.json = null;
+    } else {
       throw ex;
     }
   }

